chat-bots/src/Control/Monad/ListT.hs

```diff
@@ -41,44 +45,76 @@ newtype ListT m a = ListT
   }
 
 instance Functor m => Functor (ListT m) where
-  fmap :: Functor m => (a -> b) -> ListT m a -> ListT m b
+  fmap :: (a -> b) -> ListT m a -> ListT m b
   fmap f (ListT ma) = ListT $ fmap (bimap f (fmap f)) $ ma
 
 instance Monad m => Applicative (ListT m) where
-  pure :: Monad m => a -> ListT m a
-  pure = ListT . return . (`ConsF` emptyListT)
+  pure :: a -> ListT m a
+  pure a = a <$ Functor.introduce @_ @() ()
```

Comment:
Generally if some type constructor is a `Monad` I define its `Applicative` instance as `pure = return`, `liftA2 = liftM2`. This ensures it's impossible to get the Monad/Applicative instance coherence wrong.